packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -243,6 +247,16 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     if (!isOpen && prevState.isOpen && onMenuDismissed) {
       onMenuDismissed();
     }
+
+    if (onPendingValueSet) {
+      if (currentPendingValue !== prevState.currentPendingValue) {
+        onPendingValueSet(undefined, undefined, currentPendingValue);
+      } else if (currentPendingValueValidIndexOnHover !== prevState.currentPendingValueValidIndexOnHover) {
+        onPendingValueSet(currentOptions[currentPendingValueValidIndexOnHover], currentPendingValueValidIndexOnHover, undefined);
+      } else if (currentPendingValueValidIndex !== prevState.currentPendingValueValidIndex) {
+        onPendingValueSet(currentOptions[currentPendingValueValidIndex], currentPendingValueValidIndex, undefined);
```

Comment:
Consider pulling this into a function to keep componentDidUpdate from getting too messy

Reply:
The `else if` statements are practically the same, consider simplifying to set a local variable for the index, and if that exists then call the callback using that index